tests/pipeline/controlflowanalysis/test_readability_based_refinement.py

```diff
@@ -2077,3 +2079,121 @@ def test_declaration_listop(self, ast_call_for_loop):
         for node in ast_call_for_loop:
             if isinstance(node, ForLoopNode):
                 assert node.declaration.destination.operands[0].name == "i"
+
+    def test_skip_for_loop_recovery_if_continue_in_while(self):
+        """
+        a = 0
+        while(a < 10) {
+            if(a == 2) {
+                a = a + 2
+                continue
+            }
+            a = a + 1
+        }
+        """
+        true_value = LogicCondition.initialize_true(context := LogicCondition.generate_new_context())
+        ast = AbstractSyntaxTree(
+            root := SeqNode(true_value),
+            condition_map={
+                logic_cond("x1", context): Condition(OperationType.less, [Variable("a"), Constant(10)]),
+                logic_cond("x2", context): Condition(OperationType.equal, [Variable("a"), Constant(2)])
+            }
+        )
+
+        true_branch = ast._add_code_node(
+            [
+                Assignment(Variable("a"), BinaryOperation(OperationType.plus, [Variable("a"), Constant(2)])),
+                Continue()
+            ]
+        )
+        if_condition = ast._add_condition_node_with(logic_cond("x2", context), true_branch)
+
+        init_code_node = ast._add_code_node([Assignment(Variable("a"), Constant(0))])
+
+        while_loop = ast.factory.create_while_loop_node(logic_cond("x1", context))
+        while_loop_body = ast.factory.create_seq_node()
+        while_loop_iteration = ast._add_code_node([Assignment(Variable("a"), BinaryOperation(OperationType.plus, [Variable("a"), Constant(1)]))])
+        ast._add_node(while_loop)
+        ast._add_node(while_loop_body)
+
+        ast._add_edges_from(
+            [
+                (root, init_code_node),
+                (root, while_loop),
+                (while_loop, while_loop_body),
+                (while_loop_body, if_condition),
+                (while_loop_body, while_loop_iteration)
+            ]
+        )
+
+        WhileLoopReplacer(ast, _generate_options()).run()
+        for loop_node in list(ast.get_loop_nodes_post_order()):
+            if any(node.does_end_with_continue for node in loop_node.body.get_descendant_code_nodes_interrupting_ancestor_loop()):
+                assert isinstance(loop_node, WhileLoopNode)
+            else:
+                assert isinstance(loop_node, ForLoopNode)
```

Comment:
That is not what you want to test. 
In your test, you use more or less the same code as in the implementation. So if you would have a logical error in the implementation, the test would not fail.

Since here it is only important that no while-loop is replaced by a for-loop, you could simply check:
```python
assert not any(isinstance(loop_node, ForLoopNode) for loop_node in ast.get_loop_nodes_post_order())
```